backend/app/controllers/spree/admin/stock_locations_controller.rb

```diff
@@ -7,9 +7,10 @@ class StockLocationsController < ResourceController
 
       def set_country
         @stock_location.country = Spree::Country.default
-      rescue ActiveRecord::RecordNotFound
-        flash[:error] = Spree.t(:stock_locations_need_a_default_country)
-        redirect_to admin_stock_locations_path
+        unless @stock_location.country
```

Comment:
@aishmita-aggarwal :thought_balloon: What's the reason behind this change? The [`Spree::Country.default`](https://github.com/spree/spree/blob/master/core/app/models/spree/country.rb#L18) method will raise an exception (always) if no record is found.